common/gcp.go

```diff
@@ -3,12 +3,66 @@ package common
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
 	"net/http"
+	"os"
 
 	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/idtoken"
 	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
 )
 
+// Configures an authorizer that uses credentials sourced from JSON or file.
+// This auth mode does NOT use impersonation and it does NOT use Application
+// Default Credentials (ADC).
+func (c *DatabricksClient) configureWithGoogleCrendentials(
+	ctx context.Context) (func(r *http.Request) error, error) {
+	if c.GoogleServiceAccount == "" || c.GoogleCredentials == "" || !c.IsGcp() {
+		return nil, nil
+	}
+	if c.Host == "" {
+		return nil, fmt.Errorf("'host' field is empty or not set. "+
```

Comment:
Isn't config debug string added by Authenticate method?

Reply:
You're right. Removed.